tests/components/participants/helpers/participantDataFlow.test.ts

```diff
@@ -0,0 +1,98 @@
+import '@testing-library/jest-dom';
+import BusinessObjectNotFound from '../../../../src/common/exceptions/BusinessObjectNotFound';
+import MultipleBusinessObjectFound from '../../../../src/common/exceptions/MultipleBusinessObjectFound';
+import {
+  getParticipantByLegalName,
+  loadParticipants
+} from '../../../../src/components/participants/helpers/participantDataFlow';
+import { CypherQueryApiService as cypherQuery } from '../../../../src/services/cypherQueryApiService';
+
+import { mockParticipantsQueryResults } from './__fixtures__/participants_queryResults';
+
+// Mock the services
+jest.mock('../../../../src/services/cypherQueryApiService', () => ({
+  CypherQueryApiService: {
+    getAllParticipants: jest.fn(),
+    getParticipantByLegalName: jest.fn(),
+  }
+}));
+
+describe('Participant Data Flow', () => {
+
+  beforeEach(() => {
+    jest.clearAllMocks(); // Clear any previous mocks
+  });
+
+  describe('loadParticipants', () => {
+    it('should load participants successfully', async () => {
+      //Mock the API response
+      (cypherQuery.getAllParticipants as jest.Mock).mockResolvedValue(mockParticipantsQueryResults);
+
+      const participants = await loadParticipants();
+
+      // Assert that the correct participants are returned
+      expect(participants).toEqual(mockParticipantsQueryResults.items);
+      expect(cypherQuery.getAllParticipants).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  it('should throw an error when loading participants fails', async () => {
+    const errorMessage = 'Failed to fetch participants';
+
+    // Mock the API to reject with an error
+    (cypherQuery.getAllParticipants as jest.Mock).mockRejectedValue(new Error(errorMessage));
+
+    // Assert that an error is thrown
+    await expect(loadParticipants()).rejects.toThrow(errorMessage);
+    expect(cypherQuery.getAllParticipants).toHaveBeenCalledTimes(1);
+  });
+
+  describe('getParticipantByLegalName', () => {
+    it('should return participant details when found', async () => {
+      const mockParticipantDetail = { totalCount: 1, items: [mockParticipantsQueryResults.items[0]] };
+
+      // Mock the API response
+      (cypherQuery.getParticipantByLegalName as jest.Mock).mockResolvedValue(mockParticipantDetail);
+
+      const participant = await getParticipantByLegalName('msg systems AG');
+
+      // Assert that the correct participant details are returned
+      expect(participant).toEqual(mockParticipantDetail.items[0]);
```

Comment:
Please do not use indexed element as expected element. 

They are very fragile. If the order of the elements for some reason have to be changed then all the tests have to be fixed afterwords.  

You must not pass in the complete obejct in order to test equality. I think if some props of the participant object has given values then it can be considered correct. You can do this partial checking by using ˙expect.objectEquals(...)`

In this case it would be somehtig like:
```typescript
expect(participant).toEqual(expect.objectEquals({
    legalName: 'msg systems AG'
}));
```

This test matches only the `lealName` property.